### PR TITLE
refactor: remove secret type

### PR DESCRIPTION
**Pull request by sbchaos**

### CODE REVIEW

File: internal/store/postgres/tenant/secret_repository.go [hunk 1]
```diff
@@ -126,8 +112,8 @@ func (s SecretRepository) Save(ctx context.Context, tenantSecret *tenant.Secret)
 	}
 
 	insertSecret := `INSERT INTO secret (name, value, type, project_name, namespace_name, created_at, updated_at)
```
Comment (sravankorumilli) [inline internal/store/postgres/tenant/secret_repository.go:114]: we have type here still

File: core/tenant/secret.go [hunk 2]
```diff
@@ -8,10 +8,6 @@ const (
 	SecretStorageKey    = "STORAGE"
 	SecretSchedulerAuth = "SCHEDULER_AUTH"
 
-	// SystemDefinedSecret TODO: get rid of system defined secrets
-	SystemDefinedSecret SecretType = "system"
-	UserDefinedSecret   SecretType = "user"
-
 	SecretTypeSystemDefinedPrefix = "_OPTIMUS_"
```
Comment (sravankorumilli) [inline core/tenant/secret.go:11]: I think we are not using this prefix also, we can remove this

File: core/tenant/secret.go [hunk 3]
```diff
@@ -8,10 +8,6 @@ const (
 	SecretStorageKey    = "STORAGE"
 	SecretSchedulerAuth = "SCHEDULER_AUTH"
```
Comment (sravankorumilli) [inline core/tenant/secret.go:9]: Can you add the alert secret to be a predefined secret as well, doesn't have much significance but a place where we need what all secrets are part of infra setup.

**Review by sravankorumilli — CHANGES_REQUESTED** [review]

File: internal/store/postgres/tenant/secret_repository.go (hunk 1 above)
Comment (sbchaos) [reply internal/store/postgres/tenant/secret_repository.go:114]: Needs a migration to remove the column. Will add it.

**Review by sravankorumilli — APPROVED** [review]

**Pull request merged**